HabitTracker.harris-andy/Program.cs

```diff
@@ -0,0 +1,592 @@
+            "--------------------------------------------------\n";
+
+        int inputNumber = -1;
+        while (inputNumber < 0)
+        {
+            inputNumber = validateNumberEntry(message, isMainMenu: true);
+            Console.Clear();
+        }
+
+        switch (inputNumber)
+        {
+            case 0:
+                Console.WriteLine("\nBye!\n");
+                closeApp = true;
+                Environment.Exit(0);
+                break;
+            case 1:
+                GetAllRecords(withIds);
+                break;
+            case 2:
+                Insert();
+                break;
+            case 3:
+                Delete();
+                break;
+            case 4:
+                Update();
+                break;
+            case 5:
+                GetRecordSummary();
+                break;
+            case 6:
+                DeleteTableContents();
+                break;
+            case 7:
+                PopulateDatabase();
+                break;
+            default:
+                Console.Clear();
+                Console.WriteLine("\nInvalid Command. Give me number!");
+                break;
+        }
+    }
+}
+
+void Insert()
+{
+    Console.Clear();
+    string date = GetDateInput();
+    Console.Clear();
+    string hobby = GetHobby();
+    Console.Clear();
+    string units = GetUnitsInput();
+    Console.Clear();
+    int quantity = GetQuantityInput();
+
+    using (var connection = new SqliteConnection(connectionString))
+    {
+        connection.Open();
+        using (var command = new SqliteCommand("INSERT INTO habits (Date, Hobby, Units, Quantity) VALUES (@date, @hobby, @units, @quantity)", connection))
+        {
+            command.Parameters.AddWithValue("@date", date);
+            command.Parameters.AddWithValue("@hobby", hobby);
+            command.Parameters.AddWithValue("@units", units);
+            command.Parameters.AddWithValue("@quantity", quantity);
+
+            command.ExecuteNonQuery();
+        }
+        connection.Close();
+    }
+}
+
+string GetDateInput()
+{
+    string? date = null;
+    while (!DateTime.TryParseExact(date, format: "dd-MM-yyyy", new CultureInfo("en-US"), DateTimeStyles.None, out _))
+    {
+        Console.WriteLine("Enter date in format dd-mm-yyyy. Press 0 to return to Main Menu");
+        date = Console.ReadLine();
+        if (int.TryParse(date, out int number))
+        {
+            if (number == 0) MainMenu();
+        }
+    }
+    return date;
+}
+
+string GetUnitsInput()
+{
+    string units = "";
+
+    while (string.IsNullOrWhiteSpace(units))
+    {
+        Console.WriteLine("Enter whatever unit of measure you'd like to use. Or press 0 to return to Main Menu");
+        string? temp = Console.ReadLine();
+        if (temp != null)
+        {
+            units = temp.Trim().ToLower();
+        }
+        if (units == "0") MainMenu();
+    }
+    return units;
+}
+
+int GetQuantityInput()

```

Comment:
2 part comment. Part 2: If you accept the units `GetQuantityInput(string units)`, you could clear up your message to  `string message = $"Enter amount of {units} (greater than 0). Or press 0 to return to Main Menu";`